javalib/src/main/scala/java/util/concurrent/Semaphore.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.util.concurrent
+
+import java.util.{Collection, Collections}
+
+class Semaphore(private[this] var permits: Int, fairness: Boolean) extends Serializable {
+
+  def this(permits: Int) = this(permits, false)
+
+  // These methods can't be implemented because they block
+  // def acquire(): Unit = ???
```

Comment:
Try to be consistent with the method bodies in this list. I suggest not having them (just like abstract methods). So simply:

```scala
// def acquire(): Unit
// def acquire(permits: Int): Unit
```

etc.